File: Client/src/api/statsApi.js
class StatsApi extends Api {

	constructor(lname = "Booker", fname = "Devin", apiInterface = new AjaxCall()) {
		let url = `http://localhost:8000/stats/${lname}/${fname}`
		super(apiInterface, url)
	}

	async getData() {
		return await this.callApi()
	}
}

File: Server/stats_api.py
from api import API


class StatsAPI(API):
    def __init__(self, lname, fname):
        self.url = f"https://nba-players.herokuapp.com/players-stats/{lname}/{fname}"
        super().__init__(self.url)
        self.lname = lname
        self.fname = fname
        
    
    def proccess_data(self):
        try:
            data = self.call_api()
            return {
                "gamesPlayed": data["games_played"],
                "minutesPerGame": data["minutes_per_game"],
                "pointsPerGame": data["points_per_game"],
                "playerEfficiencyRating": data["player_efficiency_rating"]
            }

        except:
            return {
                "gamesPlayed": "Not found",
                "minutesPerGame": "Not found",
                "pointsPerGame": "Not found",
                "playerEfficiencyRating": "Not found"}
        
